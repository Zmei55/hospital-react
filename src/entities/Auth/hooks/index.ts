export * from "./useLogIn";
export * from "./useLogOut";
export * from "./useHandleChange";
