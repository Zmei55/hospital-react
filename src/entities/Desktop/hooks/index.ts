export * from "./useNewRequest";
export * from "./useFindRequest";
