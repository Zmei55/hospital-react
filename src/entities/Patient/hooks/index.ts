export * from "./useChoosePatient";
export * from "./useGetPatient";
export * from "./usePatientsListFind";
export * from "./useHandleInfoActive";
export * from "./useClearPatient";
export * from "./useTogglePatientsModal";
