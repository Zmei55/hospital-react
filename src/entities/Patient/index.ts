export * from "./components";
export * from "./redux";
export * from "./hooks";
export * from "./types";
export * from "./helpers";
export * from "./data";
