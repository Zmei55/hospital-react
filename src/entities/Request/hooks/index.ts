export * from "./useRequestServices";
export * from "./useSendRequestDetails";
