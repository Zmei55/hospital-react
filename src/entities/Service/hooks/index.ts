export * from "./useServiceAdd";
export * from "./useServiceDelete";
export * from "./useSelectedServiceDelete";
export * from "./useSelectedServiceListSave";
export * from "./useSelectedServiceListClear";
export * from "./useGetLaborsList";
export * from "./useGetServiceById";
export * from "./useGetServiceListById";
export * from "./useHandleLaborChange";
export * from "./useHandleServicesForm";
export * from "./useToggleLaborModal";
export * from "./useToggleServicesModal";
