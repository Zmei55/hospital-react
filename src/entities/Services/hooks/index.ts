export * from "./useSavedServicesList";
export * from "./useServicesFilter";
export * from "./useAddService";
