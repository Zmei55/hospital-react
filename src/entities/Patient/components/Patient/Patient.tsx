import { useState, useEffect } from "react";
import { useAppSelector } from "shared/types";
import { PatientInfo, PatientSearch, IPatient } from "entities/Patient";

import { Container } from "./Patient.styled";

const initialPatient: IPatient = {
  _id: "",
  name: "",
  birthDate: "",
  cardNumber: 0,
  gender: "",
  phoneNumber: "",
  email: "",
  identityDocument: "",
  residenceAddress: {
    street: "",
    houseNumber: "",
    city: "",
    postcode: "",
  },
  services: [],
  containers: [],
};

export function Patient() {
  const getPatient = useAppSelector(state => state.app.patient);
  const [patient, setPatient] = useState(initialPatient);

  useEffect(() => {
    if (getPatient._id !== "") setPatient(getPatient);
  }, [getPatient]);

  return (
    <Container>
      {patient._id ? <PatientInfo patient={patient} /> : <PatientSearch />}
    </Container>
  );
}
