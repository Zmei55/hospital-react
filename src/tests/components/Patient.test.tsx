import { render, screen, cleanup } from "@testing-library/react";
import { allTheProviders, patientMocks, patientListMocks } from "tests";
import * as sharedHooks from "shared/hooks";
import * as patientHooks from "entities/Patient/hooks";
import userEvent from "@testing-library/user-event";
import { PatientEl, SearchModalEl } from "entities/Patient";

afterEach(() => {
  cleanup();
});

jest.mock("shared/hooks");
jest.mock("entities/Patient/hooks");

const mockedUseSelector = jest.spyOn(sharedHooks, "useAppSelector");
const mockedUseClearPatient = jest.spyOn(patientHooks, "useClearPatient");
const mockedUseToggle = jest.spyOn(patientHooks, "useTogglePatientsModal");
const mockedUsePatientList = jest.spyOn(patientHooks, "usePatientListFind");
const mockedUseChoosePatient = jest.spyOn(patientHooks, "useChoosePatient");
const mockedUseHandleInfo = jest.spyOn(patientHooks, "useHandleInfoActive");

describe("Patient component", () => {
  it("Patient container renders", () => {
    mockedUseToggle.mockReturnValue({ togglePatientsModal: jest.fn() });

    const { container } = render(allTheProviders(<PatientEl />));

    expect(screen.getByTestId("patient-container")).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});

describe("Patient find component", () => {
  it("Patient find renders", () => {
    mockedUseToggle.mockReturnValue({ togglePatientsModal: jest.fn() });

    render(allTheProviders(<PatientEl />));

    expect(screen.getByTestId("patient-find-container")).toBeInTheDocument();
  });

  it("Patient find button renders", () => {
    mockedUseToggle.mockReturnValue({ togglePatientsModal: jest.fn() });

    render(allTheProviders(<PatientEl />));

    const findBtn = screen.getByTestId("patient-find-btn");
    expect(findBtn).toBeInTheDocument();
    expect(screen.getByText(/select a patient/i)).toBeInTheDocument();
  });
});

describe("Patient find-modal component", () => {
  it("Patient find-modal not renders", () => {
    const stateSelectors = {
      patients: {
        patient: null,
        modalPatient: false,
      },
    };
    // @ts-ignore
    mockedUseSelector.mockImplementation(state => state(stateSelectors));
    mockedUseToggle.mockReturnValue({ togglePatientsModal: jest.fn() });
    mockedUsePatientList.mockReturnValue({
      patientList: null,
      handlePatientListFind: jest.fn(),
      isLoading: false,
      isError: false,
      resetPatients: jest.fn(),
    });

    const { container } = render(allTheProviders(<PatientEl />));

    expect(screen.queryByTestId("patient-search")).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("Patient find-modal renders", () => {
    const stateSelectors = {
      patients: {
        patient: null,
        modalPatient: true,
      },
    };
    // @ts-ignore
    mockedUseSelector.mockImplementation(state => state(stateSelectors));
    mockedUseToggle.mockReturnValue({ togglePatientsModal: jest.fn() });
    mockedUsePatientList.mockReturnValue({
      patientList: null,
      handlePatientListFind: jest.fn(),
      isLoading: false,
      isError: false,
      resetPatients: jest.fn(),
    });
    mockedUseChoosePatient.mockReturnValue({ choosePatient: jest.fn() });

    const { container } = render(allTheProviders(<PatientEl />));

    expect(screen.getByTestId("patient-search")).toBeInTheDocument();

    expect(screen.getByTestId("patient-search-header")).toBeInTheDocument();
    expect(screen.getByTestId("patient-search-title")).toBeInTheDocument();
    expect(screen.getByTestId("reset-search-patient-btn")).toBeInTheDocument();
    expect(screen.getByTestId("find-search-patient-btn")).toBeInTheDocument();
    expect(screen.getByTestId("close-search-patient-btn")).toBeInTheDocument();

    expect(screen.getByTestId("patient-search-body")).toBeInTheDocument();
    expect(screen.getByTestId("patient-search-form")).toBeInTheDocument();
    expect(screen.getByTestId("patient-search-name-input")).toBeInTheDocument();
    expect(
      screen.getByTestId("patient-search-birthday-input")
    ).toBeInTheDocument();
    expect(
      screen.getByTestId("patient-search-card-number-input")
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("Patient list find", async () => {
    const handlePatientList = jest.fn();
    const stateSelectors = {
      patients: {
        patient: null,
        modalPatient: true,
      },
    };
    // @ts-ignore
    mockedUseSelector.mockImplementation(state => state(stateSelectors));
    mockedUseToggle.mockReturnValue({ togglePatientsModal: jest.fn() });
    mockedUsePatientList.mockReturnValue({
      patientList: patientListMocks,
      handlePatientListFind: handlePatientList,
      isLoading: false,
      isError: false,
      resetPatients: jest.fn(),
    });
    mockedUseChoosePatient.mockReturnValue({ choosePatient: jest.fn() });

    const { container } = render(allTheProviders(<SearchModalEl />));

    expect(screen.getByTestId("patient-search")).toBeInTheDocument();

    const patientsFindBtn = screen.getByTestId("find-search-patient-btn");

    await userEvent.click(patientsFindBtn);
    expect(handlePatientList).toHaveBeenCalledTimes(1);

    expect(screen.getByTestId("patient-search-list")).toBeInTheDocument();
    expect(
      screen.getByTestId("patient-search-list-header")
    ).toBeInTheDocument();
    expect(screen.getByText(/first name and last name/i)).toBeInTheDocument();
    expect(screen.getByText(/date of birth/i)).toBeInTheDocument();
    expect(screen.getByText(/card number/i)).toBeInTheDocument();
    expect(screen.getByTestId("patient-search-list-body")).toBeInTheDocument();

    const listitem = screen.getAllByTestId(
      "patient-search-listitem"
    )[0] as HTMLAnchorElement;
    expect(listitem).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});

describe("Patient Info component", () => {
  it("Patient Info renders - shortened", () => {
    const stateSelectors = {
      patients: {
        patient: patientMocks,
        modalPatient: false,
      },
    };
    // @ts-ignore
    mockedUseSelector.mockImplementation(state => state(stateSelectors));
    mockedUseToggle.mockReturnValue({ togglePatientsModal: jest.fn() });
    mockedUseClearPatient.mockReturnValue({ clearPatient: jest.fn() });
    mockedUseHandleInfo.mockReturnValue({
      isInfoActive: false,
      handleInfoActive: jest.fn(),
    });

    render(allTheProviders(<PatientEl />));

    expect(screen.getByTestId("patient-container")).toBeInTheDocument();
    expect(screen.getByTestId("patient-info-name-block")).toBeInTheDocument();
    expect(screen.getByTestId("more-patient-info-btn")).toBeInTheDocument();
    expect(screen.getByTestId("patient-info-name")).toBeInTheDocument();
    expect(screen.getByTestId("clear-patient-info")).toBeInTheDocument();
    expect(
      screen.getByTestId("patient-info-primary-block")
    ).toBeInTheDocument();
    expect(
      screen.getByTestId("patient-info-primary-infobox-birthday")
    ).toBeInTheDocument();
    expect(
      screen.getByTestId("patient-info-primary-infobox-gender")
    ).toBeInTheDocument();
    expect(
      screen.getByTestId("patient-info-primary-infobox-card-number")
    ).toBeInTheDocument();
    expect(
      screen.getByTestId("patient-info-primary-infobox-phone")
    ).toBeInTheDocument();
    expect(
      screen.getByTestId("patient-info-primary-infobox-email")
    ).toBeInTheDocument();
  });

  it("Patient Info renders - full", () => {
    const stateSelectors = {
      patients: {
        patient: patientMocks,
      },
    };
    // @ts-ignore
    mockedUseSelector.mockImplementation(state => state(stateSelectors));
    mockedUseToggle.mockReturnValue({ togglePatientsModal: jest.fn() });
    mockedUseClearPatient.mockReturnValue({ clearPatient: jest.fn() });
    mockedUseHandleInfo.mockReturnValue({
      isInfoActive: true,
      handleInfoActive: jest.fn(),
    });

    const { container } = render(allTheProviders(<PatientEl />));

    expect(screen.getByTestId("patient-container")).toBeInTheDocument();
    expect(screen.getByTestId("patient-info-name-block")).toBeInTheDocument();
    expect(
      screen.getByTestId("patient-info-secondary-block")
    ).toBeInTheDocument();
    expect(
      screen.getByTestId("patient-info-secondary-infobox-identity-document")
    ).toBeInTheDocument();
    expect(
      screen.getByTestId("patient-info-secondary-infobox-residence-address")
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
