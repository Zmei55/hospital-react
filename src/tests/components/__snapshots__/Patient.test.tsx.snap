// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patient Info component Patient Info renders - full 1`] = `
<div>
  <div
    class="css-1daa38u"
    data-testid="patient-container"
  >
    <div
      class="css-1273oja"
      data-testid="patient-info-name-block"
    >
      <button
        class="css-ch4pr2"
        data-testid="more-patient-info-btn"
        name="InfoButton"
        style="width: 48px; height: 48px; margin-right: 24px;"
      >
        <svg
          class="css-6vupcz"
          fill="grey"
          height="24"
          width="24"
        >
          <use
            href="svgSprite.svg#caret-double-up-bold"
          />
        </svg>
      </button>
      <h2
        class="css-1xh5jy9"
        data-testid="patient-info-name"
      >
        Emma Muster
      </h2>
      <button
        class="css-ch4pr2"
        data-testid="clear-patient-info"
        name="ClearButton"
        style="width: 48px; height: 48px;"
      >
        <svg
          class="css-nhqhda"
          fill="red"
          height="24"
          width="24"
        >
          <use
            href="svgSprite.svg#cross-bold"
          />
        </svg>
      </button>
    </div>
    <div
      class="css-h1pud"
      data-testid="patient-info-primary-block"
    >
      <div
        class="css-4xbszx"
        data-testid="patient-info-primary-infobox-birthday"
      >
        <span
          class="css-jrfass"
        >
          Date of birth
        </span>
        <span
          class="css-p6kx9h"
        >
          26
  .12
  .1968
        </span>
      </div>
      <div
        class="css-4xbszx"
        data-testid="patient-info-primary-infobox-gender"
      >
        <span
          class="css-jrfass"
        >
          Sex
        </span>
        <span
          class="css-p6kx9h"
        >
          Female
        </span>
      </div>
      <div
        class="css-4xbszx"
        data-testid="patient-info-primary-infobox-card-number"
      >
        <span
          class="css-jrfass"
        >
          Card number
        </span>
        <span
          class="css-p6kx9h"
        >
          123456789
        </span>
      </div>
      <div
        class="css-4xbszx"
        data-testid="patient-info-primary-infobox-phone"
      >
        <span
          class="css-jrfass"
        >
          Phone number
        </span>
        <span
          class="css-p6kx9h"
        >
          +491234567890
        </span>
      </div>
      <div
        class="css-4xbszx"
        data-testid="patient-info-primary-infobox-email"
      >
        <span
          class="css-jrfass"
        >
          E-mail
        </span>
        <span
          class="css-p6kx9h"
        >
          emma-muster@mail.com
        </span>
      </div>
    </div>
    <div
      class="css-1832p0z"
      data-testid="patient-info-secondary-block"
    >
      <div
        class="css-4xbszx"
        data-testid="patient-info-secondary-infobox-identity-document"
      >
        <span
          class="css-jrfass"
        >
          Identity document
        </span>
        <span
          class="css-p6kx9h"
        >
          ausweis
        </span>
      </div>
      <div
        class="css-4xbszx"
        data-testid="patient-info-secondary-infobox-residence-address"
      >
        <span
          class="css-jrfass"
        >
          Residence address
        </span>
        <span
          class="css-p6kx9h"
        >
          
            Emmentaler Stra√üe, 
            45, 
            12345, 
            Berlin
            
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Patient Search-modal component Patient Search-list find 1`] = `
<div>
  <div
    class="css-0"
    data-testid="patient-search"
  >
    <div
      class="css-1f0t3a4"
      data-testid="patient-search-header"
    >
      <h2
        class="css-w70xkr"
        data-testid="patient-search-title"
      >
        Select a patient
      </h2>
      <button
        class="css-ch4pr2"
        data-testid="reset-search-patient-btn"
        style="height: 72px; padding-right: 44px; padding-left: 44px; margin-right: 24px;"
      >
        Reset
      </button>
      <button
        class="css-mbrcjd"
        data-testid="find-search-patient-btn"
        form="patient-search-form"
        style="height: 72px; padding-right: 44px; padding-left: 44px; margin-right: 24px;"
        type="submit"
      >
        Find
      </button>
      <button
        class="css-14jjab4"
        data-testid="close-search-patient-btn"
        style="width: 72px; height: 72px;"
      >
        <svg
          class="css-bvj43j"
          fill="white"
          height="48"
          width="48"
        >
          <use
            href="svgSprite.svg#cross"
          />
        </svg>
      </button>
    </div>
    <div
      class="css-leyee6"
      data-testid="patient-search-body"
    >
      <form
        class="css-1g5v37m"
        data-testid="patient-search-form"
        id="patient-search-form"
      >
        <label
          class="css-15io445"
        >
          <input
            class="css-1ry88mf"
            data-testid="patient-search-name-input"
            name="name"
            placeholder="First name and last name"
            style="width: 100%;"
          />
        </label>
        <label
          class="css-15io445"
        >
          <input
            class="css-1ry88mf"
            data-testid="patient-search-birthday-input"
            name="birthDate"
            placeholder="Date of birth"
            style="width: 100%;"
            type="date"
          />
        </label>
        <label
          class="css-15io445"
        >
          <input
            class="css-1ry88mf"
            data-testid="patient-search-card-number-input"
            name="cardNumber"
            placeholder="Card number"
            style="width: 100%;"
          />
        </label>
      </form>
      <div
        class="css-16zoccl"
        data-testid="patient-search-list"
      >
        <ul
          class="css-135ntzi"
          data-testid="patient-search-list-header"
        >
          <li
            class="css-84kxrg"
          >
            First name and last name
          </li>
          <li
            class="css-1tbnjeu"
          >
            Date of birth
          </li>
          <li
            class="css-145tdht"
          >
            Card number
          </li>
          <li
            class="css-p9hawa"
          />
        </ul>
        <ul
          class="css-1gihon0"
          data-testid="patient-search-list-body"
        >
          <li
            class="css-hwfa6i"
            data-testid="patient-search-listitem"
          >
            <p
              class="css-1s2goei"
            >
              Emma Muster
            </p>
            <p
              class="css-13b8pju"
            >
              26.12.1968
            </p>
            <p
              class="css-1nc3dvv"
            >
              123456789
            </p>
            <p
              class="css-9xxeqo"
            >
              <button
                class="css-rpav3i"
                id="selectPatLink"
              >
                Select
              </button>
            </p>
          </li>
          <li
            class="css-hwfa6i"
            data-testid="patient-search-listitem"
          >
            <p
              class="css-1s2goei"
            >
              Ben Muster
            </p>
            <p
              class="css-13b8pju"
            >
              06.03.1970
            </p>
            <p
              class="css-1nc3dvv"
            >
              234567891
            </p>
            <p
              class="css-9xxeqo"
            >
              <button
                class="css-rpav3i"
                id="selectPatLink"
              >
                Select
              </button>
            </p>
          </li>
          <li
            class="css-hwfa6i"
            data-testid="patient-search-listitem"
          >
            <p
              class="css-1s2goei"
            >
              Bob Muster
            </p>
            <p
              class="css-13b8pju"
            >
              14.07.1973
            </p>
            <p
              class="css-1nc3dvv"
            >
              345678912
            </p>
            <p
              class="css-9xxeqo"
            >
              <button
                class="css-rpav3i"
                id="selectPatLink"
              >
                Select
              </button>
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Patient Search-modal component Patient Search-modal not renders 1`] = `
<div>
  <div
    class="css-1daa38u"
    data-testid="patient-container"
  >
    <div
      class="css-m3tvaf"
      data-testid="patient-find-container"
    >
      <button
        aria-label="find patient"
        class="css-ge3v8l"
        data-testid="patient-find-btn"
        type="button"
      >
        <svg
          class="css-smt8v5"
          height="48"
          width="48"
        >
          <use
            href="svgSprite.svg#user-plus"
          />
        </svg>
        Select a patient
      </button>
    </div>
  </div>
</div>
`;

exports[`Patient Search-modal component Patient Search-modal renders 1`] = `
<div>
  <div
    class="css-1daa38u"
    data-testid="patient-container"
  >
    <div
      class="css-m3tvaf"
      data-testid="patient-find-container"
    >
      <button
        aria-label="find patient"
        class="css-ge3v8l"
        data-testid="patient-find-btn"
        type="button"
      >
        <svg
          class="css-smt8v5"
          height="48"
          width="48"
        >
          <use
            href="svgSprite.svg#user-plus"
          />
        </svg>
        Select a patient
      </button>
    </div>
  </div>
</div>
`;

exports[`Patient component Patient container renders 1`] = `
<div>
  <div
    class="css-1daa38u"
    data-testid="patient-container"
  >
    <div
      class="css-m3tvaf"
      data-testid="patient-find-container"
    >
      <button
        aria-label="find patient"
        class="css-ge3v8l"
        data-testid="patient-find-btn"
        type="button"
      >
        <svg
          class="css-smt8v5"
          height="48"
          width="48"
        >
          <use
            href="svgSprite.svg#user-plus"
          />
        </svg>
        Select a patient
      </button>
    </div>
  </div>
</div>
`;
